### ISSUE TITLE
How to use batch > 1

### ISSUE THREAD

**Issue by pvnieo** [rusty1s/pytorch_geometric#1924]
Hi,

Thanks for the great library.

I'm implementing an neural network architecture, and I'm using the `Data` object. For now, I'm only using a batch of 1 because every element has different number of nodes (I'm using a point cloud), and I want to support a batch > 1. Do you have any recommendation on how I could use the vertical batch (batches are concatenated in the axis=0) to make parallel my code?

Thank you in advance!

**Comment by rusty1s** [comment]
You can use the PyG data loader for this (`torch_geometric.data.DataLoader`), see [here](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/mutag_gin.py) for an example.

**Comment by pvnieo** [comment]
Hi @rusty1s !
Thank you for your response!

In fact, I'm aware of the `torch_geometric.data.Dataloader` object! 

Maybe my question wasn't clear, but what I want to do, is given a batch of `n` 'example, I want to manipulate them using my network, and I want to know if there is some utils function or recommended way to do it. 

For example, if all my datapoints have the same number of nodes, I could just reshape them to `batch_size x n_point x n_features`, but given the fact that each element has a different number of points, the batch is collated vertically `n_point_ver x features` where `n_point_ver = batch_size x n_point`, and I want to know how to apply my network to each datapoint individually!

Thank you in advance!

**Comment by rusty1s** [comment]
In case you want to convert your batches nodes to a representation with distinct batch dimension, you may want to look at the `torch_geometric.utils.to_dense_batch` function. This will convert your batch to `[batch_size, max_num_nodes, num_features]`.

**Comment by smorad** [comment]
For others who read this, you likely want to use the `Batch.to_dict` to get the node-to-batch indicies

```
# Results in shape [graph, max_nodes_per_graph, node_features]
torch_geometric.utils.to_dense_batch(batch.x, batch.to_dict()['batch'])
```
